src/main/java/com/minecolonies/client/gui/WindowTownHall.java

```diff
@@ -221,8 +290,76 @@ public WindowTownHall(BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton(BUTTON_UP, this::updatePriority);
+        registerButton(BUTTON_DOWN, this::updatePriority);
+        registerButton(BUTTON_DELETE, this::deleteWorkOrder);
     }
 
+    /**
+     * On Button click update the priority.
+     * @param button the clicked button.
+     */
+    private void updatePriority(Button button)
+    {
+        Label idLabel = (Label)button.getParent().getChildren().get(HIDDEN_ID_POSITION);
+        int id = Integer.parseInt(idLabel.getLabelText());
+        String buttonLabel = button.getID();
+
+        for (int i = 0; i < workOrders.size(); i++)
+        {
+            WorkOrderView workOrder = workOrders.get(i);
+            if (workOrder.getId() == id)
+            {
+                if(buttonLabel.equals(BUTTON_UP))
+                {
+                    if (i > 0)
+                    {
+                        WorkOrderView workOrderUp = workOrders.get(i-1);
+                        workOrder.setPriority(workOrderUp.getPriority()+1);
+
+                        MineColonies.getNetwork().sendToServer(new WorkOrderChangeMessage(this.building, id, false , workOrderUp.getPriority()+1));
+                    }
+                }
+                else if (buttonLabel.equals(BUTTON_DOWN))
+                {
+                    if (i <= workOrders.size())
+                    {
+                        WorkOrderView workOrderDown = workOrders.get(i+1);
+
+                        workOrder.setPriority(workOrderDown.getPriority()-1);
+                        MineColonies.getNetwork().sendToServer(new WorkOrderChangeMessage(this.building, id, false , workOrderDown.getPriority()-1));
+                    }
+                }
+                workOrders.remove(i);
+                workOrders.add(workOrder);
+                break;
+
+            }
+        }
+        Collections.sort(workOrders, (first, second) -> second.getPriority() > first.getPriority() ? 1 : (second.getPriority() < first.getPriority() ? -1 : 0));
+        window.findPaneOfTypeByID(LIST_WORKORDER, ScrollingList.class).refreshElementPanes();
+    }
+
+    /**
+     * On Button click remove the workOrder.
+     * @param button the clicked button.
+     */
+    private void deleteWorkOrder(Button button)
+    {
+        //todo player has permission to do this?
+        Label idLabel = (Label)button.getParent().getChildren().get(HIDDEN_ID_POSITION);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'idLabel' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)